Compiler/script2/cs_compiler.cpp

```diff
@@ -201,6 +201,12 @@ static std::unique_ptr<ScriptTOC> ccCompileDataTOC(const SymbolTable &symt, cons
         new ScriptTOC(std::move(tocb.Finalize(rtti))));
 }
 
+#ifdef __GNUC__
+inline int strncpy_s(char* dest, size_t destsz, const char* src, size_t count)
+{
+    strncpy(dest, src, std::min(destsz, count)); return 0;
+}
+#endif
```

Comment:
The replacements for standard string functions are better put into string_compat.h/cpp.
strncpy_s seems to be a standard now, starting with C11, according to https://en.cppreference.com/w/c/string/byte/strncpy

Reply:
C11 is not C++11, you can't access it from a C++ code unit.

Reply:
Yes I know that C11 and C++11 are different things. What I meant is that's a part of a standard C library, and C library may be used by C++ compilers.

I am not aware of such details whether C11 standard is widely supported by C++ compilers now, but, for example, C99 is already supported by most C++11 compatible compilers.

Reply:
I wrote one and placed in the string_compat